### PR TITLE
Updates to GC rooting doc

### LINKED ISSUE #68: GC Rooting guide should indicate that typedefs are no longer preferred
Inside the engine we've decided to deprecate support for the typedefs; probably should update the GC rooting guide to share that guidance as well. (I doubt they'll go away, but should perhaps avoid them)

### PR DESCRIPTION
**Pull request by hotsphink**
Minor updates to current conventions, with some small clarifications and grammar touchups.

### CODE REVIEW

File: docs/GC Rooting Guide.md [hunk 1]
```diff
@@ -398,8 +352,8 @@ JS::PersistentRooted<MyOwningStruct> immortalStruct;
 But note that `JS::PersistentRooted` in a struct or class is a rather
 dangerous thing to use — it will keep a GC thing alive, and most GC
 things end up keeping their global alive, so if your class/struct is
-reachable in any way from that global, then nothing will ever be cleaned
-up by the GC.
+reachable in any way from that global, then nothing reachable form that
```
Comment (mgaudet) [inline docs/GC Rooting Guide.md]: *from

**Review by mgaudet — APPROVED** [review]
LGTM

File: docs/GC Rooting Guide.md [hunk 2]
```diff
@@ -17,12 +17,12 @@ The main types of GC thing pointer are:
...
 - `JSScript*`
-- `jsid`
+- `JS::PropertyKey (aka jsid)
 - `JSFunction*`
 - `JS::Symbol*`
 
-Note that `JS::Value` and `jsid` can contain pointers internally even
-though they are not a normal pointer type, hence their inclusion in this
+Note that `JS::Value` and `JS::PropertyKey` can contain pointers internally even
+though they are normal pointer types, hence their inclusion in this
```
Comment (redfire75369) [inline docs/GC Rooting Guide.md:25]: I don't think they're normal pointer types. Seems like the "not" was accidentally removed.

Comment (hotsphink) [reply docs/GC Rooting Guide.md:25]: Oops, thanks. Fixed.

**Pull request merged**